instrumentation/opentelemetry-instrumentation-aio-pika/src/opentelemetry/instrumentation/aio_pika/span_builder.py

```diff
@@ -49,18 +49,30 @@ def set_destination(self, destination: str):
         self._attributes[SpanAttributes.MESSAGING_DESTINATION] = destination
 
     def set_channel(self, channel: AbstractChannel):
-        url = channel.connection.connection.url
-        self._attributes.update({
-            SpanAttributes.NET_PEER_NAME: url.host,
-            SpanAttributes.NET_PEER_PORT: url.port
-        })
+        connection = channel.connection
```

Comment:
Notice that this change shouldn't be necessary - the constructor of `Exchange` doesn't get the connection as its first argument anymore (https://github.com/mosquito/aio-pika/commit/c2ee4be6e4f254e0d36c99f5bcd906cf31fa8233). If you take that into consideration, this change would not be necessary

Reply:
@nozik I'm not sure I'm following, could you be a bit more specific with what you would think this `set_channel` method needs to change to if not what I have here?

Reply:
Ah I see, over in https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1605/files#diff-500514cf8be14e4864fdf49daa4c761c0b1ad0c358cec84b4e26a2c62c1c5dffR57.

That change you made there is incomplete. The change I made here is necessary for aiopika 8.x, the call here fails otherwise:

```python
url = channel.connection.connection.url
```

In your PR, you are still using an aio_pika v7 mock that assumes a property path from `channel.connection.connection.url`. In aio_pika v8, the extra `.connection` is removed, and it's just `channel.connection.url`.

The change occurred actually in the same commit you linked, [these lines](https://github.com/mosquito/aio-pika/commit/c2ee4be6e4f254e0d36c99f5bcd906cf31fa8233#diff-34571b521f08738cafd0fc3ecf1d653376a9f979e8ad04982ff7814a46fcd617L545-R626)

Reply:
@phillipuniverse Notice [here](https://github.com/mosquito/aio-pika/commit/c2ee4be6e4f254e0d36c99f5bcd906cf31fa8233) the changes to `exchange.py` - the `Exchange` class no longer receives `AbstractConnection` as its first constructor argument. However, in `test_get_publish_span` we're still sending wrongfully sending it as the first argument. If you just check the condition of the `aio-pika` version and adapt the creation of the `Exchange` (simply remove the first ctor argument for version >= 8.0.0), all the other code changes will become redundant. I hope this helps.

Reply:
There are 2 changes that needed to happen to support aio_pika 8:

1. The change you are describing to the constructor in the test as aio_pika 8 changed the signature
2. The change in this method to deal with the changed shape of AbstractConnection

There aren’t good integration tests for aio_pika yet in this repo but I observed the problem in this method in my integration environment after patching the instrumentation version check for to indiscriminately apply to aio_pika 8

Reply:
👍

Reply:
Hi there,
I'm also looking for support of aio_pika 8 so I wonder, when optimistically this PR could be merged and released? Thank you in advance.

Reply:
This instrumentation is supported and maintained by community members. Maintainers don't have any expertise in this instrumentation to judge the changes. The approval from the [component owner](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/.github/component_owners.yml) gets it merged immediately and will be part of the next scheduled release.